### PR TITLE
AArch64 centos docker image ks file

### PR DESCRIPTION
**Pull request by jefby**
create centos7.1 docker image in AArch64,and it can work well , https://hub.docker.com/r/jefby/centos-arm64/

### CODE REVIEW

**Comment by jperrin** [comment]
Thanks for putting this together. Out of curiosity, why change the path in img2tar to be /tmp/${image} as opposed to ${image} ?

**Comment by jperrin** [comment]
I'll be putting together a new build of golang and an updated docker binary at some point today if you wouldn't mind testing them.

**Comment by jefby** [comment]
@jperrin Very pleasure.because it can't generate a tar.bz2 file in local directory when i run this scripts.

File: docker/centos-arm64.ks [hunk 1]
```diff
@@ -0,0 +1,134 @@
+install
+keyboard us
+network  --bootproto=dhcp --device=eth0 --onboot=on
+rootpw --iscrypted $1$UKLtvLuY$kka6S665oCFmU7ivSDZzU.
+timezone UTC --isUtc 
+selinux --enforcing
+firewall --disabled
+#repo --name="CentOS" --baseurl=http://mirror.centos.org/centos/7/os/x86_64/ --cost=100
+#repo --name="CentOS" --baseurl=ftp://192.168.3.1/centos7forARM/   --cost=100
```
Comment (stpierre) [inline docker/centos-arm64.ks]: Why are these left in?

File: docker/centos-arm64.ks [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+repo --name="CentOS" --baseurl=http://buildlogs.centos.org/centos/7/os/aarch64/   --cost=100
+# here you must download from my shared rpm package,and create a localrepo . and MAKEDEV rpm package
+repo --name="systemdcontainer" --baseurl=ftp://192.168.3.1/systemd-container/ --cost=100
```
Comment (stpierre) [inline docker/centos-arm64.ks]: Having an RFC1918 address here seems undesirable.

File: docker/centos-arm64.ks [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+#  ldconfig
+rm -rf /etc/ld.so.cache
+rm -rf /var/cache/ldconfig/*
+rm -rf /var/cache/yum/* 
+
+# Clean up after the installer.
+rm -f /etc/rpm/macros.imgcreate
+
+# temp fix for systemd /run/lock
```
Comment (stpierre) [inline docker/centos-arm64.ks]: What does this comment refer to?

Comment (jperrin) [inline docker/centos-arm64.ks]: An old one. There some other stuff in here I need to clean up, but it's a good initial start.

File: docker/img2tar.sh [hunk 4]
```diff
@@ -22,7 +22,8 @@ cd "$mount"
 #tar -cpSf - --acls --selinux --xattrs * | bzip2 > ${image}.tar.bz2
 
 # This one appears to work fine for docker creation
-tar --numeric-owner -c . | bzip2 > ${image}.tar.bz2
+# hacked by jefby,change the output file to /tmp/xxx.tar.bz2
+tar --numeric-owner -c . | bzip2 > /tmp/${image}.tar.bz2
```
Comment (stpierre) [inline docker/img2tar.sh]: 1. Why is this necessary?

2. This makes it impossible to use img2tar.sh with a qualified (fully or otherwise) path, e.g.:

    img2tar.sh foo/bar.img
    img2tar.sh /tmp/foo/img

3. If this really honestly needs to go in /tmp then you need to use a proper tmpfile (``mktemp``) to avoid exposing a known-tempfile attack.

File: docker/centos-arm64.ks (hunk 2 above)
Comment (jefby) [inline docker/centos-arm64.ks]: Because build aarch64 image needs systemd-container rpm package , and there are no package in any website , and i compile it locally , so if you need to build the docker aarch64 image , you must createrepo in local and set the repository.

**Comment by jperrin** [comment]
@stpierre if you want to fix up img2tar with a mktemp that's fine. I'm changing the kickstart currently to reflect proper mirrors and to match more closely what we do with the current 7 kickstart. There's also no 'systemd-container built for aarch64 yet, which is something I should address.

**Comment by jperrin** [comment]
@jefby For arm64/aarch64 it's quite a bit easier to use livemedia-creator rather than ami_creator, which is rather old. The only reason I've not migrated *everything* to use livemedia-creator or similar tools is because it doesn't support CentOS 5.
